app/models/query/results.rb

```diff
@@ -50,8 +50,8 @@ def work_packages
 
   def sorted_work_packages_matching_the_filters_today
     sorted_work_packages
-      .visible
       .merge(filtered_work_packages.merge(filter_merges))
+      .visible
```

Comment:
Interesting... `merge` was overriding filter conditions set by `visible`?

Reply:
It does when you add a where condition for the same field. Before the move to WP based permissions, the `visible` scope added a where on the `project_id` field. The move to WP based permissions moved the `where` condition to `id`. But the `filter_merges` query already has a bunch of where conditions on the `id` field, so it overrides those when merging.

In the [docs for #merge](https://api.rubyonrails.org/classes/ActiveRecord/SpawnMethods.html#method-i-merge) it warns about this:

> For conditions that exist in both relations, those from other will take precedence. To find the intersection of two relations, use `QueryMethods#and`.


```ruby
[1] pry(main)> WorkPackage.where(id: [1,2,3]).merge(WorkPackage.where(id: 5)).to_sql
=> "SELECT \"work_packages\".* FROM \"work_packages\" WHERE \"work_packages\".\"id\" = 5"

[2] pry(main)> WorkPackage.where(id: [1,2,3]).merge(WorkPackage.where(project_id: 1)).to_sql
=> "SELECT \"work_packages\".* FROM \"work_packages\" WHERE \"work_packages\".\"id\" IN (1, 2, 3) AND \"work_packages\".\"project_id\" = 1"

[3] pry(main)> WorkPackage.where(id: [1,2,3]).and(WorkPackage.where(id: 5)).to_sql
=> "SELECT \"work_packages\".* FROM \"work_packages\" WHERE \"work_packages\".\"id\" IN (1, 2, 3) AND \"work_packages\".\"id\" = 5"
```

Reply:
Great debugging. Thank you